### DISCUSSION TITLE
nerdctl does not respect '--ip' option while running a container

### DISCUSSION THREAD

**Discussion by mangudaigb** [containerd/nerdctl#1528] [Q&A] [answered]
I am trying to assign a static ip to a container, but I am getting the following error:
```
stderr: time="2022-11-21T07:18:37Z" level=fatal msg="containerIP assignment is not supported in rootless mode"
```
I believe this should not be the case as IP assignment should be allowed in rootless mode.

## Steps to reproduce
1. Create a network: lima nerdctl network create -d bridge --subnet 201.10.0.0/28 mynetwork
2. Create a mysql container: lima nerdctl run --name mysql-master --net mysqlnetwork --ip 201.10.0.6 -e MYSQL_ROOT_PASSWORD=password -v /Users/<user>/containers/mysql-master/conf.d:/etc/mysql/conf.d -d mysql:latest

**Result**: The container is created with a arbitrary IP from the subnet that has been specified but not with the IP I want.

**Log**:
FATA[0000] failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: error during container init: error running hook #0: error running hook: exit status 1, stdout: , stderr: time="2022-11-21T07:18:37Z" level=fatal msg="containerIP assignment is not supported in rootless mode"
Failed to write to log, write /home/lima.linux/.local/share/nerdctl/1935db59/containers/default/97b93ea3f409e6459b33e36d8ffca8ccd5913952b078f53100d73c0ebe7347b5/oci-hook.createRuntime.log: file already closed: unknown

Am I doing something wrong or is this the expected behavior?

Thanks

**Comment by AkihiroSuda** [comment, marked as answer]
Seems a bug. 

Being fixed in:
- https://github.com/containerd/nerdctl/pull/1529 (Target: nerdctl v1.1.0)

**Reply by mangudaigb** [reply]
Ahh! Thanks. :)
